### ISSUE TITLE
(cdk-diff): >=2.11.0 not authorized to perform: cloudformation:ListStackResources

### ISSUE THREAD

**Issue by natevick** [aws/aws-cdk#19351]
### General Issue

cdk-[QUALIFIER]-deploy-role-[ACCOUNT]-[REGION]/aws-cdk-node is not authorized to perform: cloudformation:ListStackResources

### The Question

Prior to 2.11.0 running diff  just worked without errors or abnormal output. Starting at 2.11.0 and all the way through 2.15.0 I get `cdk-[QUALIFIER]-deploy-role-[ACCOUNT]-[REGION]/aws-cdk-node is not authorized to perform: cloudformation:ListStackResources`. 

The bootstrap version is 10. 
There is a lookup role in IAM.
The trusted account and trusted lookup account is the primary/only account involved in the project.
The project was started at 2.1.0.

I'm not sure what to ask at this point, but I'm currently stuck at 2.10.0.

### CDK CLI Version

2.11.0

### Node.js Version

16.13.1

### OS

Debian Bullseye 11.1

### Language

Typescript

### Language Version

4.5.2

### Other information

I did note this [merged PR](https://github.com/aws/aws-cdk/pull/18277) but I believe that came into CDK at 2.7.0.

**Comment by groner** [comment]
I'm also seeing this error with `cdk diff` (I didn't try deploy) starting in 2.10.0. The cdk bootstrap version here is also 10.

**Comment by peterwoodworth** [comment]
Wonder if it was [this PR?](https://github.com/aws/aws-cdk/pull/18207)

**Comment by comcalvi** [comment]
Thanks for the issue @nvick @groner, could you provide a reproduction stack?

**Comment by nvick** [comment]
```typescript
import { Stack, StackProps } from 'aws-cdk-lib'
import { Construct } from 'constructs'
import * as ec2 from 'aws-cdk-lib/aws-ec2'
import * as eks from 'aws-cdk-lib/aws-eks'

export class CdkStackDiffIssuesStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const cluster = new eks.Cluster(this, 'staging-eks', {
      clusterName: 'staging',
      version: eks.KubernetesVersion.V1_21,
      defaultCapacity: 0,
      albController: {
        version: eks.AlbControllerVersion.V2_3_0
      }
    })

    cluster.addNodegroupCapacity('staging-node-group', {
      instanceTypes: [new ec2.InstanceType('t3.large')],
      minSize: 1,
      maxSize: 10
    })
  }
}
```
If I run `cdk diff` with the above stack on version 2.10.0 no errors, but if I run it with 2.11.0-2.16.0 I get the error.

**Comment by nvick** [comment]
I just verified with my reproduction stack it is only `diff` I have been able to reproduce.

**Comment by nvick** [comment]
@comcalvi I also pushed this to a repo with a devcontainer configuration. https://github.com/nvick/cdk-stack-diff-issues

**Comment by nvick** [comment]
Sure @comcalvi!

- Fork above repo
- Start Codespace
- Run `aws configure`
- Run `npm install`
- Run `cdk diff` note full diff output
- Run `npm install -g aws-cdk@2.11.0`
- Run `cdk diff` note error
- Run `npm install -g aws-cdk`
- Run `cdk diff` not error

**Comment by comcalvi** [comment]
Thanks @nvick this reproduces the error. I'll be investigating this issue to determine root cause later.

**Comment by comcalvi** [comment]
@nvick The source of the error is the version of `aws-cdk-lib` in your `package.json`. In here:

```
  "devDependencies": {
    // ...
    "aws-cdk": "^2.11.0",
    // ...
  },
  "dependencies": {
    "aws-cdk-lib": "2.3.0",
     // ...
  }
```

if you change `2.3.0` to `2.11.0` and run `npm install`, then the error no longer occurs. I'm leaving this issue open so we can fix the error message, but that should unblock you.

**Comment by nvick** [comment]
Thanks @comcalvi! I'll try it and let you know if I run into anything else. Thanks for taking the time.

**Comment by comcalvi** [comment]
Any version pair that satisfies this condition:

```
    "aws-cdk" >= "2.11.0"
    "aws-cdk-lib" <= "2.6.0"
```

causes the bug. It's because #18207 (introduced in 2.11.0) depends on #18277 (introduced in 2.7.0). The consequence is that `aws-cdk` >= 2.11.0 depend on `aws-cdk-lib` >= 2.7.0.

**Comment by toduythienluong** [comment]
> I'm also seeing this error with `cdk diff` (I didn't try deploy) starting in 2.10.0. The cdk bootstrap version here is also 10.

I also got with cdk diff, how to fix it?

**Comment by ricardosllm** [comment]
I also had this issue, fixed in the interim by adding the policy to the role explicitly, in this case `AWSCloudFormationReadOnlyAccess`

**Comment by toduythienluong** [comment]
> I also had this issue, fixed in the interim by adding the policy to the role explicitly, in this case `AWSCloudFormationReadOnlyAccess`

thanks, that fixes

**Comment by aarondodd** [comment]
Same, CDK v2 doesn't let me do `cdk diff` until I add `AWSCloudFormationReadOnlyAccess` to the role. Deploy has been working fine, though.

**Comment by jhkcia** [comment]
I also have this problem with last version of aws-cdk

**Comment by mrgrain** [comment]
This has long been resolved in recent version of CDK. Make sure you update to a recent version for all bugfixes. Also ensure you are using the latest bootstrap version. Closing this here. If anyone is still encountering the issue with up-to-date versions, please open a new issue.
